code/entities/machine/soda-fountain/SodaFountain.CupFillable.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class CupFillable : ModelEntity
+{
+    public enum MaterialGroup : int
```

Comment:
`MaterialGroup` is a somewhat internal / implementation based idea.

For more readable and sustainable code use names which are more related to the purpose of the variable/type/enum/class/struct.

You use this enum to indicate the "Type" of soda, blue, green, red, black, default, and then based on that type, create an item of that sode "type"

Don't publicly expose the material group enum, expose the "type of soda" enum.

Ie, make this `SodaType` not `MaterialGroup`.

And suggestion is to remove the "Default" type, you could name it "Regular".

Reply:
Note: I see that you already have a `SodaType` in `SodaFountain`, that was excellent! You should be using that same type here

Reply:
I changed this to CupColor and put it into the main SodaFountain class. My current thinking is to have a separate type for soda and cup color to make it more flexible (i.e. cup color can be random instead of determined by soda type)